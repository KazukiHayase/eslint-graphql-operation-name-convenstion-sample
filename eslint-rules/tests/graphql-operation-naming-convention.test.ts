import { join } from "path";
import rule from "../graphql-operation-naming-convention";
import { GraphQLRuleTester } from "@graphql-eslint/eslint-plugin";

const TEST_SCHEMA = /* GraphQL */ `
  type Query {
    todos: [Todo!]!
  }

  type Todo {
    id: Int!
    text: String!
    done: Boolean!
  }
`;

const WITH_SCHEMA = {
  parserOptions: {
    schema: TEST_SCHEMA,
  },
};

const ruleTester = new GraphQLRuleTester();

ruleTester.runGraphQLTests("graphql-operation-naming-convention", rule, {
  valid: [
    {
      ...WITH_SCHEMA,
      filename: join(__dirname, "TodoList.graphql"),
      code: "query TodoListQuery { todos { id } }",
    },
    {
      ...WITH_SCHEMA,
      filename: join(__dirname, "TodoList/index.graphql"),
      code: "query TodoListQuery { todos { id } }",
    },
    {
      ...WITH_SCHEMA,
      filename: join(__dirname, "TodoList.graphql"),
      code: "query TodoListComponentQuery { todos { id } }",
    },
    {
      ...WITH_SCHEMA,
      filename: join(__dirname, "TodoItem.graphql"),
      code: "fragment TodoItem_Todo on Todo { id }",
    },
    {
      ...WITH_SCHEMA,
      filename: join(__dirname, "TodoItem/index.graphql"),
      code: "fragment TodoItem_Todo on Todo { id }",
    },
    {
      ...WITH_SCHEMA,
      filename: join(__dirname, "TodoItem.graphql"),
      code: "fragment TodoItemComponent_Todo on Todo { id }",
    },
  ],
  invalid: [
    {
      ...WITH_SCHEMA,
      filename: join(__dirname, "TodoList.graphql"),
      code: "query TodoList { todos { id } }",
      errors: [
        {
          message: `Operation "TodoList" should have prefix "TodoList" and suffix "Query"`,
        },
      ],
    },
    {
      ...WITH_SCHEMA,
      filename: join(__dirname, "TodoList.graphql"),
      code: "query TodosQuery { todos { id } }",
      errors: [
        {
          message: `Operation "TodosQuery" should have prefix "TodoList" and suffix "Query"`,
        },
      ],
    },
    {
      ...WITH_SCHEMA,
      filename: join(__dirname, "TodoItem.graphql"),
      code: "fragment TodoItem on Todo { id }",
      errors: [
        {
          message: `Fragment "TodoItem" should have prefix "TodoItem" and suffix "_Todo"`,
        },
      ],
    },
    {
      ...WITH_SCHEMA,
      filename: join(__dirname, "TodoItem.graphql"),
      code: "fragment Todos_Todo on Todo { id }",
      errors: [
        {
          message: `Fragment "Todos_Todo" should have prefix "TodoItem" and suffix "_Todo"`,
        },
      ],
    },
  ],
});
