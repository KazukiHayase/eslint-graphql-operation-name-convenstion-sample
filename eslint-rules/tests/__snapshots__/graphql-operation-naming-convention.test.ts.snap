// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Invalid #1 1`] = `
"#### ⌨️ Code

      1 | query TodoList { todos { id } }

#### ❌ Error

    > 1 | query TodoList { todos { id } }
        |       ^^^^^^^^ Operation \\"TodoList\\" should have prefix \\"TodoList\\" and suffix \\"Query\\"

#### 💡 Suggestion: Rename to \`TodoListQuery\`

    1 | query TodoListQuery { todos { id } }"
`;

exports[`Invalid #2 1`] = `
"#### ⌨️ Code

      1 | query TodosQuery { todos { id } }

#### ❌ Error

    > 1 | query TodosQuery { todos { id } }
        |       ^^^^^^^^^^ Operation \\"TodosQuery\\" should have prefix \\"TodoList\\" and suffix \\"Query\\"

#### 💡 Suggestion: Rename to \`TodoListQuery\`

    1 | query TodoListQuery { todos { id } }"
`;

exports[`Invalid #3 1`] = `
"#### ⌨️ Code

      1 | fragment TodoItem on Todo { id }

#### ❌ Error

    > 1 | fragment TodoItem on Todo { id }
        |          ^^^^^^^^ Fragment \\"TodoItem\\" should have prefix \\"TodoItem\\" and suffix \\"_Todo\\"

#### 💡 Suggestion: Rename to \`TodoItem_Todo\`

    1 | fragment TodoItem_Todo on Todo { id }"
`;

exports[`Invalid #4 1`] = `
"#### ⌨️ Code

      1 | fragment Todos_Todo on Todo { id }

#### ❌ Error

    > 1 | fragment Todos_Todo on Todo { id }
        |          ^^^^^^^^^^ Fragment \\"Todos_Todo\\" should have prefix \\"TodoItem\\" and suffix \\"_Todo\\"

#### 💡 Suggestion: Rename to \`TodoItem_Todo\`

    1 | fragment TodoItem_Todo on Todo { id }"
`;
